pkgs/stdenv/generic/check-meta.nix

```diff
@@ -42,8 +42,7 @@ let
   allowUnsupportedSystem = config.allowUnsupportedSystem or false
     || builtins.getEnv "NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM" == "1";
 
-  isUnfree = licenses: lib.lists.any (l:
-    !l.free or true || l == "unfree" || l == "unfree-redistributable") licenses;
+  isUnfree = licenses: lib.lists.any (l: !l.free or true) licenses;
```

Comment:
I may be reading this incorrectly, but as far as I can tell both `unfree` and `unfreeRedistributable` have `free = false`, which means the first condition will always return true and we don't need the special casing:
https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix#L600-L608

Reply:
I think this comes from a while back when licenses were just strings - not attrsets. Might be reasonable to just leave it though.

Reply:
It's somewhat reasonable, I suppose, but it's dead code -- and the stdenv package is already quite big and complicated.  I'd argue we should take the opportunity to simplify our logic slightly.

Reply:
We can just flat up require they not be strings elsewhere too

Reply:
Seems like a sensible improvement.  I did some initial searching with 

```
git grep 'license = ".*'
```

and found a few thousand packages with string licenses, but only in ~160 files.  Looks like we can fix the bulk of them by updating the Haskell and NodeJS generators to output attr licenses instead of string licenses.